File: lab-1/domain/playlist_manager.py
from content_manager import ContentManager


class PlaylistManager(ContentManager):
    def __init__(self, name):
        super().__init__(name)
        self.playlists = []

    def add_item(self, playlist):
        self.playlists.append(playlist)

    def remove_item(self, playlist):
        if playlist in self.playlists:
            self.playlists.remove(playlist)

    def display_contents(self):
        for playlist in self.playlists:
            playlist.display_info()

File: src/main/kotlin/content_manager/implementations/PlaylistManager.kt
package content_manager.implementations
import content_manager.ContentManager
import media_item.MediaItem
import media_item.implementations.Playlist

class PlaylistManager(override val name: String) : ContentManager {
    val playlists = mutableListOf<Playlist>()

    override fun addItem(item: MediaItem) {
        if (item is Playlist) {
            playlists.add(item)
        }
    }

    override fun removeItem(item: MediaItem) {
        playlists.remove(item)
    }

    override fun displayContents() {
        playlists.forEach { it.displayInfo() }
    }
}